TestB2Storage is consistently failing in post-merge builds

```
--- FAIL: TestB2Storage (66.41s)
    retriable_t.go:95: Sleeping 2s before running attempt #2
    retriable_t.go:95: Sleeping 4s before running attempt #3
    retriable_t.go:95: Sleeping 8s before running attempt #4
    retriable_t.go:95: Sleeping 10s before running attempt #5
    --- FAIL: TestB2Storage/Attempt-5 (15.34s)
        asserts.go:104: GetMetadata() failed: bad_request: Bad fileId: test-1590990725-b17c26484aa9625e-abcdbbf4f0507d054ed5a80a5b65086f602b
        asserts.go:108: invalid blob ID on abcdbbf4f0507d054ed5a80a5b65086f602b: , want abcdbbf4f0507d054ed5a80a5b65086f602b
        asserts.go:117: invalid timestamp on abcdbbf4f0507d054ed5a80a5b65086f602b: 0001-01-01 00:00:00 +0000 UTC, want 2020-06-01 05:52:06 +0000 UTC
        asserts.go:104: GetMetadata() failed: bad_request: Bad fileId: test-1590990725-b17c26484aa9625e-abff4585856ebf0748fd989e1dd623a8963d
        asserts.go:108: invalid blob ID on abff4585856ebf0748fd989e1dd623a8963d: , want abff4585856ebf0748fd989e1dd623a8963d
        asserts.go:112: invalid length on abff4585856ebf0748fd989e1dd623a8963d: 0, want 1000
        asserts.go:117: invalid timestamp on abff4585856ebf0748fd989e1dd623a8963d: 0001-01-01 00:00:00 +0000 UTC, want 2020-06-01 05:52:07 +0000 UTC
        asserts.go:104: GetMetadata() failed: bad_request: Bad fileId: test-1590990725-b17c26484aa9625e-abgc3dca496d510f492c858a2df1eb824e62
        asserts.go:108: invalid blob ID on abgc3dca496d510f492c858a2df1eb824e62: , want abgc3dca496d510f492c858a2df1eb824e62
        asserts.go:112: invalid length on abgc3dca496d510f492c858a2df1eb824e62: 0, want 10000
        asserts.go:117: invalid timestamp on abgc3dca496d510f492c858a2df1eb824e62: 0001-01-01 00:00:00 +0000 UTC, want 2020-06-01 05:52:08 +0000 UTC
        asserts.go:104: GetMetadata() failed: bad_request: Bad fileId: test-1590990725-b17c26484aa9625e-kopia.repository
        asserts.go:108: invalid blob ID on kopia.repository: , want kopia.repository
        asserts.go:112: invalid length on kopia.repository: 0, want 100
        asserts.go:117: invalid timestamp on kopia.repository: 0001-01-01 00:00:00 +0000 UTC, want 2020-06-01 05:52:09 +0000 UTC
        asserts.go:104: GetMetadata() failed: bad_request: Bad fileId: test-1590990725-b17c26484aa9625e-zxce0e35630770c54668a8cfb4e414c6bf8f
        asserts.go:108: invalid blob ID on zxce0e35630770c54668a8cfb4e414c6bf8f: , want zxce0e35630770c54668a8cfb4e414c6bf8f
        asserts.go:112: invalid length on zxce0e35630770c54668a8cfb4e414c6bf8f: 0, want 1
        asserts.go:117: invalid timestamp on zxce0e35630770c54668a8cfb4e414c6bf8f: 0001-01-01 00:00:00 +0000 UTC, want 2020-06-01 05:52:07 +0000 UTC
        asserts.go:104: GetMetadata() failed: bad_request: Bad fileId: test-1590990725-b17c26484aa9625e-abcdbbf4f0507d054ed5a80a5b65086f602b
        asserts.go:108: invalid blob ID on abcdbbf4f0507d054ed5a80a5b65086f602b: , want abcdbbf4f0507d054ed5a80a5b65086f602b
        asserts.go:117: invalid timestamp on abcdbbf4f0507d054ed5a80a5b65086f602b: 0001-01-01 00:00:00 +0000 UTC, want 2020-06-01 05:52:06 +0000 UTC
        asserts.go:104: GetMetadata() failed: bad_request: Bad fileId: test-1590990725-b17c26484aa9625e-abff4585856ebf0748fd989e1dd623a8963d
        asserts.go:108: invalid blob ID on abff4585856ebf0748fd989e1dd623a8963d: , want abff4585856ebf0748fd989e1dd623a8963d
        asserts.go:112: invalid length on abff4585856ebf0748fd989e1dd623a8963d: 0, want 1000
        asserts.go:117: invalid timestamp on abff4585856ebf0748fd989e1dd623a8963d: 0001-01-01 00:00:00 +0000 UTC, want 2020-06-01 05:52:07 +0000 UTC
        asserts.go:104: GetMetadata() failed: bad_request: Bad fileId: test-1590990725-b17c26484aa9625e-abgc3dca496d510f492c858a2df1eb824e62
        asserts.go:108: invalid blob ID on abgc3dca496d510f492c858a2df1eb824e62: , want abgc3dca496d510f492c858a2df1eb824e62
        asserts.go:112: invalid length on abgc3dca496d510f492c858a2df1eb824e62: 0, want 10000
        asserts.go:117: invalid timestamp on abgc3dca496d510f492c858a2df1eb824e62: 0001-01-01 00:00:00 +0000 UTC, want 2020-06-01 05:52:08 +0000 UTC
        asserts.go:104: GetMetadata() failed: bad_request: Bad fileId: test-1590990725-b17c26484aa9625e-abff4585856ebf0748fd989e1dd623a8963d
        asserts.go:108: invalid blob ID on abff4585856ebf0748fd989e1dd623a8963d: , want abff4585856ebf0748fd989e1dd623a8963d
        asserts.go:112: invalid length on abff4585856ebf0748fd989e1dd623a8963d: 0, want 1000
        asserts.go:117: invalid timestamp on abff4585856ebf0748fd989e1dd623a8963d: 0001-01-01 00:00:00 +0000 UTC, want 2020-06-01 05:52:12 +0000 UTC
        asserts.go:104: GetMetadata() failed: bad_request: Bad fileId: test-1590990725-b17c26484aa9625e-abgc3dca496d510f492c858a2df1eb824e62
        asserts.go:108: invalid blob ID on abgc3dca496d510f492c858a2df1eb824e62: , want abgc3dca496d510f492c858a2df1eb824e62
        asserts.go:112: invalid length on abgc3dca496d510f492c858a2df1eb824e62: 0, want 10000
        asserts.go:117: invalid timestamp on abgc3dca496d510f492c858a2df1eb824e62: 0001-01-01 00:00:00 +0000 UTC, want 2020-06-01 05:52:13 +0000 UTC
        asserts.go:104: GetMetadata() failed: bad_request: Bad fileId: test-1590990725-b17c26484aa9625e-abff4585856ebf0748fd989e1dd623a8963d
        asserts.go:108: invalid blob ID on abff4585856ebf0748fd989e1dd623a8963d: , want abff4585856ebf0748fd989e1dd623a8963d
        asserts.go:112: invalid length on abff4585856ebf0748fd989e1dd623a8963d: 0, want 1000
        asserts.go:117: invalid timestamp on abff4585856ebf0748fd989e1dd623a8963d: 0001-01-01 00:00:00 +0000 UTC, want 2020-06-01 05:52:12 +0000 UTC
        asserts.go:104: GetMetadata() failed: bad_request: Bad fileId: test-1590990725-b17c26484aa9625e-abgc3dca496d510f492c858a2df1eb824e62
        asserts.go:108: invalid blob ID on abgc3dca496d510f492c858a2df1eb824e62: , want abgc3dca496d510f492c858a2df1eb824e62
        asserts.go:112: invalid length on abgc3dca496d510f492c858a2df1eb824e62: 0, want 10000
        asserts.go:117: invalid timestamp on abgc3dca496d510f492c858a2df1eb824e62: 0001-01-01 00:00:00 +0000 UTC, want 2020-06-01 05:52:13 +0000 UTC
        asserts.go:104: GetMetadata() failed: bad_request: Bad fileId: test-1590990725-b17c26484aa9625e-kopia.repository
        asserts.go:108: invalid blob ID on kopia.repository: , want kopia.repository
        asserts.go:112: invalid length on kopia.repository: 0, want 100
        asserts.go:117: invalid timestamp on kopia.repository: 0001-01-01 00:00:00 +0000 UTC, want 2020-06-01 05:52:14 +0000 UTC
        asserts.go:104: GetMetadata() failed: bad_request: Bad fileId: test-1590990725-b17c26484aa9625e-zxce0e35630770c54668a8cfb4e414c6bf8f
        asserts.go:108: invalid blob ID on zxce0e35630770c54668a8cfb4e414c6bf8f: , want zxce0e35630770c54668a8cfb4e414c6bf8f
        asserts.go:112: invalid length on zxce0e35630770c54668a8cfb4e414c6bf8f: 0, want 1
        asserts.go:117: invalid timestamp on zxce0e35630770c54668a8cfb4e414c6bf8f: 0001-01-01 00:00:00 +0000 UTC, want 2020-06-01 05:52:12 +0000 UTC
FAIL
```

--- a/repo/blob/b2/b2_storage.go
+++ b/repo/blob/b2/b2_storage.go
@@ -80,11 +80,31 @@ func (s *b2Storage) GetBlob(ctx context.Context, id blob.ID, offset, length int6
 	return v.([]byte), nil
 }
 
+func (s *b2Storage) resolveFileID(fileName string) (string, error) {
+	resp, err := s.bucket.ListFileVersions(fileName, "", 1)
+	if err != nil {
+		return "", err
+	}
+
+	if len(resp.Files) > 0 {
+		if resp.Files[0].Name == fileName && resp.Files[0].Action == backblaze.Upload {
+			return resp.Files[0].ID, nil
+		}
+	}
+
+	return "", nil
+}
+
 func (s *b2Storage) GetMetadata(ctx context.Context, id blob.ID) (blob.Metadata, error) {
 	fileName := s.getObjectNameString(id)
 
 	attempt := func() (interface{}, error) {
-		fi, err := s.bucket.GetFileInfo(fileName)
+		fileID, err := s.resolveFileID(fileName)
+		if err != nil {
+			return nil, err
+		}
+
+		fi, err := s.bucket.GetFileInfo(fileID)
 		if err != nil {
 			return nil, err
 		}

